errors.go

```diff
@@ -109,6 +124,40 @@ func sprintf(format string, args ...interface{}) string {
 	return fmt.Sprintf(format, args...)
 }
 
+// NewBatch creates a new BatchError.
+func NewBatch() BatchError {
+	return &batcherrtype{}
+}
+
+//Append appends the error
+func (b *batcherrtype) Append(err error) {
+	b.errors = append(b.errors, err)
+}
+
+//Errors returns the errors
+func (b *batcherrtype) Errors() []error {
+	return b.errors
+}
+
+//IsEmpty returns if there is any error
+func (b *batcherrtype) IsEmpty() bool {
+	return len(b.errors) == 0
+}
```

Comment:
Just to be sure I understood the idea.

So currently a usage of the library will be:

	func ChainCall(f *functions, arg) error {
		errs := errors.NewBatch()
	
		for _, transform := range f.functions {
			err := transform.Apply(arg)
			if err != nil {
				errs.Append(err)
			}
		}
	
		if !errs.IsEmpty() {
			return errs
		}
	
		return nil
	}
	

By requiring an error or an array of errors one example of an usage of this library will be:	
	
	func ChainCall(f *functions, arg) error {
		var errorStack []error

		for _, transform := range f.functions {
			err := transform.Apply(arg)
			if err != nil {
				errorStack.append(err)
			}
		}
	
		if !errs.IsEmpty() {
			return newBatch(errorStack)
		}
	
		return nil
	}